channels/routing.py

```diff
@@ -87,7 +87,14 @@ def __init__(self, routes):
             # The inner ASGI app wants to do additional routing, route
             # must not be an endpoint
             if getattr(route.callback, "_path_routing", False) is True:
-                route.pattern._is_endpoint = False
+                pattern = route.pattern
+                if isinstance(pattern, RegexPattern):
+                    arg = pattern._regex
+                elif isinstance(pattern, RoutePattern):
+                    arg = pattern._route
+                else:
+                    raise ValueError(f"Unsupported pattern type: {type(pattern)}")
+                route.pattern = pattern.__class__(arg, pattern.name, is_endpoint=False)
```

Comment:
using `pattern.__class__` to support subclasses, just in case someone is using them...